script.js

```diff
@@ -0,0 +1,141 @@
+            }
+            fetch(`https://www.googleapis.com/youtube/v3/videos?key=AIzaSyB0Yem1_WTQl_SDGS1Sxkb46EnEKQiV5vI&id=${ids.join(',')}&part=snippet,statistics`)
+                .then(videoDetails => videoDetails.json())
+                .then(videoDetails => {
+                    for (let i in searchResults.items)
+                        videoDetails.items[i].snippet.shortDesc = searchResults.items[i].snippet.description;
+                    getVid(videoDetails.items)
+                });
+        });
+//     fetch("./search.json")
+//         .then(searchResults => searchResults.json())
+//         .then(searchResults => {
+//             let ids = []
+//             for (let video of searchResults.items) {
+//                 ids.push(video.id.videoId);
+//             }
+//             fetch("./searchvide.json")
+//                 .then(videoDetails => videoDetails.json())
+//                 .then(videoDetails => {
+//                     for (let i in searchResults.items)
+//                         videoDetails.items[i].snippet.shortDesc = searchResults.items[i].snippet.description;
+//                     getVid(videoDetails.items)
+//                 });
+//         });
+ });
+
+
+
+
+
+
+function getVid(videoDetails) {
+    let i = 1
+    for (let video of videoDetails) {
+        fillDetails(video, i++);
+    }
+    hideAll();
+    displayVideos(1)
+}
+
+window.onresize = function () {
+    // console.log(document.querySelectorAll(".active")[0].getAttribute("id"));
+    displayVideos(document.querySelectorAll(".active")[0].getAttribute("id").substring(4));
+}
+
+function fillDetails(details, i) {
+    // console.log(details);
+    let video = document.importNode(template, true);
+    video.querySelector(".ytVideo").setAttribute("id", "video" + i);
+    video.querySelector(".title").textContent = details.snippet.title;
+    video.querySelector(".views").textContent = details.statistics.viewCount + " views";
+    video.querySelector(".published").textContent = details.snippet.publishedAt.substring(0, 10).split("-").join("/");
+    video.querySelector(".author").textContent = details.snippet.channelTitle;
+    video.querySelector(".author").setAttribute("onclick", "window.open('https://www.youtube.com/channel/" + details.snippet.channelId + "')");
+    video.querySelector(".description").textContent = details.snippet.shortdescription;
+
+
+    // video.querySelector(".vid").setAttribute("src", "https://www.youtube.com/embed/" + details.id);
+    video.querySelector(".thumbnail").src = details.snippet.thumbnails.high.url;
+    video.querySelector(".thumbnail").setAttribute("onclick", "window.open('https://www.youtube.com/watch?v=" + details.id + "')");
+
+    // video.querySelector("#video" + i).setAttribute("onclick", "window.open('https://www.youtube.com/watch?v=" + details.id + "')");
+
+    document.querySelector(".main").appendChild(video);
+}
+
+
+function hideAll() {
+    for (let i = 1; i <= 15; i++) {
```

Comment:
please don't use magic numbers here and also the index variable names should not be i,j
you can avoid index variables by using a forEach loop.